src/utils/nodeUtils.ts

```diff
@@ -70,3 +70,47 @@ export function redirect(
     history.push(url);
   }
 }
+
+const getAncestorNodes = (
+  pathway: Pathway,
+  rootNodeKey: string,
+  currNodeKey: string
+): Array<PathwayNode> => {
+  let ancestors: Array<PathwayNode> = [];
+  let previousNodes: Array<PathwayNode> = [];
+  const currNode = pathway.nodes[currNodeKey];
+
+  currNode.transitions.forEach(transition => {
+    if (transition.transition === rootNodeKey) ancestors.push(currNode);
+    else {
+      previousNodes = getAncestorNodes(pathway, rootNodeKey, transition.transition);
+      if (previousNodes?.length) {
+        if (!ancestors.some(node => node.key === currNode.key)) ancestors.push(currNode);
+        ancestors = ancestors.concat(previousNodes);
+      }
+    }
+  });
+
+  return ancestors;
+};
+
+export const getConnectableNodes = (
+  pathway: Pathway,
+  rootNode: PathwayNode
+): Array<{ label: string; value: string }> => {
+  const connectableNodes: Array<{ label: string; value: string }> = [];
+  const ancestorNodes = getAncestorNodes(pathway, rootNode.key ?? '', 'Start');
+  ancestorNodes.push(rootNode);
+
+  Object.keys(pathway.nodes).forEach(nodeKey => {
+    const node = pathway.nodes[nodeKey];
+    const rootNodeConnectsToNode = rootNode.transitions.some(
+      transition => transition.transition === nodeKey
+    );
+    const nodeIsAncestor = ancestorNodes.some(node => node.key === nodeKey);
+    if (!rootNodeConnectsToNode && !nodeIsAncestor && node.key)
```

Comment:
It looks like we aren't allowing transitions to ancestor nodes. Is this what we want? This prevents loops, but I wasn't sure if we wanted to prevent those.

Reply:
I was also unsure about this. I think it should be a different task to research/make a decision about this because I see how loops would be useful, but loops are also only useful if the clinician can tell that a patient is on the loop for the x time - otherwise it would just appear that they're in that node on the pathway for the first time. Currently we don't store the information to figure out if someone is in a loop / what # cycle in the loop so there is no way to display it, meaning the loop wouldn't give us any information that undoing or advancing a patient back and forth across two nodes would.